### PR TITLE
tree and snapshots do not need closing

### PR DESCRIPTION
**Pull request by gsserge**
If we ever need to do anything special when `Tree` and `Snapshot` are closed, then we will implement the manual drop logic using [Drop::drop](https://doc.rust-lang.org/std/ops/trait.Drop.html).

### CODE REVIEW

File: src/snapshot.rs [hunk 1]
```diff
@@ -133,15 +95,11 @@ impl<P: KeyTrait, V: Clone> Snapshot<P, V> {
     }
 
     pub fn iter(&self) -> Result<Iter<P, V>, TrieError> {
-        // Check if the snapshot is already closed
-        self.is_closed()?;
         Ok(Iter::new(self.root.as_ref()))
```
Comment (arriqaaq) [inline src/snapshot.rs:98]: WDYT of getting rid of these Result too in this PR?

File: src/snapshot.rs [hunk 2]
```diff
@@ -133,15 +95,11 @@ impl<P: KeyTrait, V: Clone> Snapshot<P, V> {
...
     }
 
     /// Returns an iterator over all versions of the key-value pairs within the Trie.
     pub fn iter_with_versions(&self) -> Result<VersionedIter<P, V>, TrieError> {
-        // Check if the snapshot is already closed
-        self.is_closed()?;
         Ok(VersionedIter::new(self.root.as_ref()))
```
Comment (arriqaaq) [inline src/snapshot.rs:103]: Same here! I think the removal of Result in methods like these will help

File: src/snapshot.rs (hunk 1 above)
Comment (gsserge) [reply src/snapshot.rs:98]: Good point! Will do 👍

Comment (gsserge) [reply src/snapshot.rs:98]: Let me handle this in a follow-up removing `Result`'s altogether.

**Review by arriqaaq — APPROVED** [review]
LGTM!

**Pull request merged**